src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporter.cs

```diff
@@ -476,52 +474,91 @@ internal void TestCompleted(
 
         terminal.AppendLine();
 
-        FormatErrorMessage(terminal, errorMessage);
+        FormatErrorMessage(terminal, errorMessage, exception, outcome);
         FormatExpectedAndActual(terminal, expected, actual);
-        FormatStackTrace(terminal, errorStackTrace);
+        FormatStackTrace(terminal, exception);
+        FormatInnerExceptions(terminal, exception);
         FormatStandardAndErrorOutput(terminal, standardOutput, errorOutput);
     }
 
-    private static void AppendAssemblyLinkTargetFrameworkAndArchitecture(ITerminal terminal, string assembly, string? targetFramework, string? architecture)
+    private static void FormatInnerExceptions(ITerminal terminal, Exception? exception)
     {
-        terminal.AppendLink(assembly, lineNumber: null);
-        if (targetFramework != null || architecture != null)
+        IEnumerable<Exception?> aggregateExceptions = exception switch
         {
-            terminal.Append(" (");
-            if (targetFramework != null)
-            {
-                terminal.Append(targetFramework);
-                terminal.Append('|');
-            }
+            AggregateException aggregate => aggregate.Flatten().InnerExceptions,
+            _ => [exception?.InnerException],
+        };
 
-            if (architecture != null)
+        foreach (Exception? aggregate in aggregateExceptions)
+        {
+            Exception? currentException = aggregate;
+            while (currentException is not null)
             {
-                terminal.Append(architecture);
+                terminal.SetColor(TerminalColor.Red);
+                terminal.Append(SingleIndentation);
+                terminal.Append("--->");
+                FormatErrorMessage(terminal, null, currentException, TestOutcome.Error);
+
+                FormatStackTrace(terminal, currentException);
+
+                currentException = currentException.InnerException;
             }
+        }
+    }
 
-            terminal.Append(')');
+    private static void FormatErrorMessage(ITerminal terminal, string? errorMessage, Exception? exception, TestOutcome outcome)
+    {
+        if (RoslynString.IsNullOrWhiteSpace(errorMessage) && exception is null)
+        {
+            return;
         }
+
+        terminal.SetColor(TerminalColor.Red);
+
+        if (exception is null)
+        {
+            AppendIndentedLine(terminal, errorMessage, SingleIndentation);
+        }
+        else if (outcome == TestOutcome.Fail)
+        {
+            // For failed tests, we don't prefix the message with the exception type as it's most likely some assertion failure.
```

Comment:
```suggestion
            // For failed tests, we don't prefix the message with the exception type because it is most likely an assertion specific exception like AssertionFailedException, and we prefer to show that without the exception type to avoid additional noise.
```

Reply:
Just my opinion here, but I actually don't like that the exception type is missing, and find it useful to see it.

From my experience, test failures can be due to flakiness (e.g. network, timeouts, etc) and the exception type makes it much quicker and easier to realise what the error is. And it's only ever something like 20 characters long so I wouldn't consider it noise.

Reply:
If your test adapter reports `TestOutcome.Error` you will go in the else branch where we do display the exception full name as prefix. As explained in the comment, most of the time this is linked to assertion failure where just displaying the message is clearer.

I would prefer to start with that version and collect some evidence of what our users like/don't like rather than making a decision based on a few of us.